relayer/chains/cosmos/feegrant.go

```diff
@@ -373,110 +400,96 @@ func (cc *CosmosProvider) GrantAllGranteesBasicAllowance(ctx context.Context, ga
 
 // GrantBasicAllowance Send a feegrant with the basic allowance type.
 // This function does not check for existing feegrant authorizations.
-// TODO: check for existing authorizations prior to attempting new one.
 func (cc *CosmosProvider) GrantAllGranteesBasicAllowanceWithExpiration(ctx context.Context, gas uint64, expiration time.Time) error {
 	if cc.PCfg.FeeGrants == nil {
-		return errors.New("ChainClient must be a FeeGranter to establish grants")
+		return errors.New("chain client must be a FeeGranter to establish grants")
 	} else if len(cc.PCfg.FeeGrants.ManagedGrantees) == 0 {
-		return errors.New("ChainClient is a FeeGranter, but is not managing any Grantees")
+		return errors.New("chain client is a FeeGranter, but is not managing any Grantees")
 	}
 
-	granterKey := cc.PCfg.FeeGrants.GranterKey
+	granterKey := cc.PCfg.FeeGrants.GranterKeyOrAddr
 	if granterKey == "" {
 		granterKey = cc.PCfg.Key
 	}
 
 	granterAddr, err := cc.GetKeyAddressForKey(granterKey)
 	if err != nil {
-		fmt.Printf("ChainClient FeeGranter misconfiguration: %s", err.Error())
+		cc.log.Error("Unknown granter", zap.String("key name", granterKey))
```

Comment:
nit: log field names should be snake_case https://github.com/cosmos/relayer/blob/main/docs/logging.md#usage